Packs/CommonScripts/Scripts/SetAndHandleEmpty/SetAndHandleEmpty.py

```diff
@@ -1,16 +1,35 @@
 import demistomock as demisto
 from CommonServerPython import *
+from typing import Text
+
+
+def get_value(value, stringify=False):
+    if stringify:
+        return str(value)
+    elif not value or not isinstance(value, (Text, bytes, bytearray)):
+        return value
+    else:
+        try:
+            return json.loads(value)
+        except (json.decoder.JSONDecodeError, TypeError):
+            return value
 
 
 def main():
     args = demisto.args()
     value = args.get('value')
+    key = args.get('key')
+    value = get_value(value, args.get('stringify') == 'true')
     if value:
-        human_readable = 'Key ' + args.get('key') + ' set'
-        context_entry = {args.get('key'): value}
+        human_readable = f'Key {key} set'
+        context_entry = {key: value}
     else:
-        context_entry = {}
         human_readable = 'value is None'
+        context_entry = {}
+
+    if args.get('append') == 'false' and context_entry:
+        demisto.executeCommand('DeleteContext', {'key': key})
```

Comment:
why `Delete`?

Reply:
Because when append is false we should override the old key.